### PR TITLE
feat: integrating a new tool called jina search

### PR DESCRIPTION
**Pull request by Yash-1511**
# Pull Request Description

This pull request integrates a new tool called `jina` into our platform, addressing the limitation of LLMs regarding their fixed knowledge cutoff. `Jina search` provides dynamic grounding capabilities by accessing the latest information from the web, thus enhancing the factuality and relevance of responses. 

Fixes # (issue not specified)

## New Package?

- [x] Yes
- [ ] No

## Version Bump?

- [x] Yes
- [ ] No

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/tools/llama-index-tools-jina/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# LlamaIndex Tools Integration: Jina
+
```
Comment (logan-markewich) [inline llama-index-integrations/tools/llama-index-tools-jina/README.md:2]: nit: lets add the install here: `pip install llama-index-tools-jina` ?

File: llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import requests
+from yarl import URL
+from typing import Dict, List
+from llama_index.core.schema import Document
+from llama_index.core.tools.tool_spec.base import BaseToolSpec
+
+JINA_SEARCH_URL_ENDPOINT = "https://s.jina.ai/"
+
+
+class JinaToolSpec(BaseToolSpec):
```
Comment (logan-markewich) [inline llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py:10]: oh neat, this doesn't require an API key?

File: llama-index-integrations/tools/llama-index-tools-jina/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# Feel free to un-skip examples, and experimental, you will just need to
+# work through many typos (--write-changes and --interactive will help)
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = false
+import_path = "llama_index.tools.jina"
+
+[tool.llamahub.class_authors]
+CLASS = "Yash-1511"
```
Comment (logan-markewich) [inline llama-index-integrations/tools/llama-index-tools-jina/pyproject.toml:17]: Lets update this with the real class name

File: llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        return response.json()
+
+    def jina_search(self, query: str) -> List[Document]:
+        """
+        Make a query to the Jina Search engine to receive a list of results.
+
+        Args:
+            query (str): The query to be passed to Jina Search.
+            search_lang (str): The search language preference (ISO 639-1), default is "en".
+            num_results (int): The number of search results returned in response, default is 5.
```
Comment (logan-markewich) [inline llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py:42]: we've listed 3 args here, but the function only accepts query? Is this a mistake?

File: llama-index-integrations/tools/llama-index-tools-jina/README.md (hunk 1 above)
Comment (Yash-1511) [reply llama-index-integrations/tools/llama-index-tools-jina/README.md:2]: > nit: lets add the install here: `pip install llama-index-tools-jina` ?

Added.

File: llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py (hunk 2 above)
Comment (Yash-1511) [reply llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py:10]: Where i mentioned that it requires API key?

Comment (logan-markewich) [reply llama-index-integrations/tools/llama-index-tools-jina/llama_index/tools/jina/base.py:10]: You didn't -- I was just surprised that it a service like this works without one

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**